pcsx2-qt/MainWindow.cpp

```diff
@@ -1581,14 +1571,12 @@ void MainWindow::onChangeDiscMenuAboutToHide()
 
 void MainWindow::onLoadStateMenuAboutToShow()
 {
-	if (m_save_states_invalidated)
-		updateSaveStateMenus(m_current_disc_path, m_current_game_serial, m_current_game_crc);
+	updateSaveStateMenus(m_current_disc_path, m_current_game_serial, m_current_game_crc);
```

Comment:
If I was doing it, I'd remove the menu populating logic from `updateSaveStateMenus()`, and call `populateLoadStateMenu()`/`populateSaveStateMenu()` in these event handlers. Saves a few stat calls when booting games.